src/include/wiredtiger.in

```diff
@@ -4802,7 +4817,38 @@ struct __wt_storage_source {
 	    WT_LOCATION_HANDLE *location_handle, const char *name, bool *existp);
 
 	/*!
-	 * Open a handle for a named storage source object
+	 * Flush any existing objects that match the location and name from
+	 * local storage to shared object storage.  The implementation guarantees
+	 * that all objects that are in a created state (see WT_STORAGE_SOURCE::ss_open_object)
+	 * at the beginning of this call have been transferred when this call returns.
+	 *
+	 * @errors
+	 *
+	 * @param storage_source the WT_STORAGE_SOURCE
+	 * @param session the current WiredTiger session
+	 * @param location_handle the location to flush (or NULL for all)
+	 * @param name the name of the object to flush (or NULL for all)
+	 * @param config additional configuration, currently must be NULL
+	 */
+	int (*ss_flush)(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
+	    WT_LOCATION_HANDLE *location_handle, const char *name, const char *config);
+
+	/*!
+	 * Open a handle for a named storage source object.
+	 *
+	 * Objects created are not deemed to "exist" and be visible to other APIs
+	 * like WT_STORAGE_SOURCE::ss_exist until the new handle has been closed.
+	 * Objects once created are immutable. That is, only objects that do not already
+         * exist can be opened with the create flag, and objects that already exist can
+         * only be opened with the readonly flag.
```

Comment:
There is some unexpected spacing going on at the beginning of this line. Any idea why?

Reply:
clang format doesn't run on wiredtiger.in.  Much of wiredtiger.in uses tabs, and I need to remember to do this - sometimes I don't remember...  It might be nice to 1) convert all tabs to spaces in wiredtiger.in 2) teach clang-format about it, which might be challenging.

Reply:
Fixed.

Reply:
Maybe a tabs vs. spaces thing?